examples/compression/src/main.rs

```diff
@@ -0,0 +1,286 @@
+use axum::{routing::post, Json, Router};
+use serde_json::Value;
+use tower::ServiceBuilder;
+use tower_http::{compression::CompressionLayer, decompression::RequestDecompressionLayer};
+use tracing_subscriber::{layer::SubscriberExt, util::SubscriberInitExt};
+
+#[tokio::main]
+async fn main() {
+    tracing_subscriber::registry()
+        .with(
+            tracing_subscriber::EnvFilter::try_from_default_env()
+                .unwrap_or_else(|_| "compression=trace".into()),
+        )
+        .with(tracing_subscriber::fmt::layer())
+        .init();
+
+    let app: Router = app();
+
+    let listener = tokio::net::TcpListener::bind("127.0.0.1:3000")
+        .await
+        .unwrap();
+    tracing::debug!("listening on {}", listener.local_addr().unwrap());
+    axum::serve(listener, app).await.unwrap();
+}
+
+fn app() -> Router {
+    Router::new().route("/", post(root)).layer(
+        ServiceBuilder::new()
+            .layer(RequestDecompressionLayer::new())
+            .layer(CompressionLayer::new()),
+    )
+}
+
+async fn root(Json(value): Json<Value>) -> Json<Value> {
+    Json(value)
+}
+
```

Comment:
Can you add a visual delimiter here like `// ===== TESTS =====` so that it is clearer for the reader that anything bellow is superfluous for the example they were looking for?

Reply:
https://github.com/tokio-rs/axum/pull/2623/commits/8efe6f1b73c4c8562fd4772c02410849c8044076

Reply:
Do we have a marker like that elsewhere? Seems a bit weird to me honestly.

Reply:
I don't think so, but we also don't have tests in examples so there really never was a need.

 But personally, when I skimmed through the code, I suddenly found myself between the `#[test]` decorated functions and it wasn't as easy for me to visually find where the code ends and the tests begin since.

But I don't mind if you don't like this and you'd rather just have it as it was.

Reply:
I could also move the tests to a separated `test.rs`. WDYT?

Reply:
That sounds like a good idea! 👍

Reply:
https://github.com/tokio-rs/axum/pull/2623/commits/179fff01852679bafc7324aad242161fb1dbe651